net/newsboat/Portfile

```diff
@@ -2,395 +2,33 @@
-                    sha256  49f2bd0c6468a8230e1db229cff8029217cf623c767ea5d60bfbd42729ea54d5 \
-                    size    32228 \
-                    unicode-normalization-0.1.7.crate \
-                    rmd160  4cbee8c9b2979aa0aa93f5a222fd0b72c46e97c6 \
-                    sha256  6a0180bc61fc5a987082bfa111f4cc95c4caff7f9799f3e46df09163a937aa25 \
-                    size    330545 \
-                    unicode-width-0.1.5.crate \
-                    rmd160  360df9e831a6e20931c240d13747f3711dc568d9 \
-                    sha256  882386231c45df4700b275c7ff55b6f3698780a650026380e72dabe76fa46526 \
-                    size    15761 \
-                    unicode-xid-0.1.0.crate \
-                    rmd160  fc5a8141e55bf6e2660b8c588e1107f179d24bb8 \
-                    sha256  fc72304796d0818e357ead4e000d19c9c174ab23dc11093ac919054d20a6a7fc \
-                    size    16000 \
-                    unreachable-1.0.0.crate \
-                    rmd160  00c79ce6e523b3b09978db8766e3110a637c23ec \
-                    sha256  382810877fe448991dfc7f0dd6e3ae5d58088fd0ea5e35189655f84e6814fa56 \
-                    size    6355 \
-                    url-1.7.2.crate \
-                    rmd160  c46442b7903a874b0556861845d5121bfc3b5397 \
-                    sha256  dd4e7c0d531266369519a4aa4f399d748bd37043b00bde1e4ff1f60a120b355a \
-                    size    68597 \
-                    utf8-ranges-1.0.2.crate \
-                    rmd160  472c5b94c5ca826c8788d2e6629b713a9df70fd6 \
-                    sha256  796f7e48bef87609f7ade7e06495a87d5cd06c7866e6a5cbfceffc558a243737 \
-                    size    8510 \
-                    version_check-0.1.5.crate \
-                    rmd160  0806190559062dc843ecf13393f6c1319367eac1 \
-                    sha256  914b1a6776c4c929a602fafd8bc742e06365d4bcbe48c30f9cca5824f70dc9dd \
-                    size    8173 \
-                    void-1.0.2.crate \
-                    rmd160  5d76f91beb625f5b645c156ca45ee5138e984e80 \
-                    sha256  6a02e4885ed3bc0f2de90ea6dd45ebcbb66dacffe03547fadbb0eeae2770887d \
-                    size    2356 \
-                    winapi-0.3.6.crate \
-                    rmd160  549ef0ff7cd1a0f2aabf915ae603ed2c3c920ab6 \
-                    sha256  92c1eb33641e276cfa214a0522acad57be5c56b10cb348b3c5117db75f3ac4b0 \
-                    size    1029391 \
-                    winapi-i686-pc-windows-gnu-0.4.0.crate \
-                    rmd160  a7d1e9e7f940d2e376a1b6dede7f0a50ad191ab8 \
-                    sha256  ac3b87c63620426dd9b991e5ce0329eff545bccbbb34f3be09ff6fb6ab51b7b6 \
-                    size    2918815 \
-                    winapi-x86_64-pc-windows-gnu-0.4.0.crate \
-                    rmd160  300417853d251d91cadb9650992a6aa98248619f \
-                    sha256  712e227841d057c1ee1cd2fb22fa7e5a5461ae8e48fa2ca79ec42cfc1931183f \
-                    size    2947998
+github.tarball_from archive
 
 depends_build       port:asciidoc \
```

Comment:
I'm the maintainer of Newsboat. Just wanted to let you know that we no longer use Asciidoc — it was completely replaced by Asdiidoctor.

Reply:
Hello @Minoru, thank you for letting me know...

...would you have any inside into the cause of the build failure?:
```
libtool: compile:  /usr/bin/clang -c -DLOCALEDIR=\"/opt/local/var/macports/build/_Users_runner_work_1_s_net_newsboat/newsboat/work/newsboat-r2.20.1/target/x86_64-apple-darwin/release/build/gettext-sys-a078d908ca433957/out/share/locale\" -DLOCALE_ALIAS_PATH=\"/opt/local/var/macports/build/_Users_runner_work_1_s_net_newsboat/newsboat/work/newsboat-r2.20.1/target/x86_64-apple-darwin/release/build/gettext-sys-a078d908ca433957/out/share/locale\" -DLIBDIR=\"/opt/local/var/macports/build/_Users_runner_work_1_s_net_newsboat/newsboat/work/newsboat-r2.20.1/target/x86_64-apple-darwin/release/build/gettext-sys-a078d908ca433957/out/lib\" -DBUILDING_LIBINTL -DBUILDING_DLL -DIN_LIBINTL -DENABLE_RELOCATABLE=1 -DIN_LIBRARY -DINSTALLDIR=\"/opt/local/var/macports/build/_Users_runner_work_1_s_net_newsboat/newsboat/work/newsboat-r2.20.1/target/x86_64-apple-darwin/release/build/gettext-sys-a078d908ca433957/out/lib\" -DNO_XMALLOC -Dset_relocation_prefix=libintl_set_relocation_prefix -Drelocate=libintl_relocate -DDEPENDS_ON_LIBICONV=1 -DHAVE_CONFIG_H -I. -I/opt/local/var/macports/build/_Users_runner_work_1_s_net_newsboat/newsboat/work/newsboat-r2.20.1/target/x86_64-apple-darwin/release/build/gettext-sys-a078d908ca433957/out/gettext/gettext-runtime/intl -I.. -I/opt/local/include -isysroot/Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk -O3 -ffunction-sections -fdata-sections -fPIC -Os -isysroot/Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk -arch x86_64 --target=x86_64-apple-darwin -fvisibility=hidden /opt/local/var/macports/build/_Users_runner_work_1_s_net_newsboat/newsboat/work/newsboat-r2.20.1/target/x86_64-apple-darwin/release/build/gettext-sys-a078d908ca433957/out/gettext/gettext-runtime/intl/xsize.c -o xsize.o
libtool: compile:  /usr/bin/clang -c -DLOCALEDIR=\"/opt/local/var/macports/build/_Users_runner_work_1_s_net_newsboat/newsboat/work/newsboat-r2.20.1/target/x86_64-apple-darwin/release/build/gettext-sys-a078d908ca433957/out/share/locale\" -DLOCALE_ALIAS_PATH=\"/opt/local/var/macports/build/_Users_runner_work_1_s_net_newsboat/newsboat/work/newsboat-r2.20.1/target/x86_64-apple-darwin/release/build/gettext-sys-a078d908ca433957/out/share/locale\" -DLIBDIR=\"/opt/local/var/macports/build/_Users_runner_work_1_s_net_newsboat/newsboat/work/newsboat-r2.20.1/target/x86_64-apple-darwin/release/build/gettext-sys-a078d908ca433957/out/lib\" -DBUILDING_LIBINTL -DBUILDING_DLL -DIN_LIBINTL -DENABLE_RELOCATABLE=1 -DIN_LIBRARY -DINSTALLDIR=\"/opt/local/var/macports/build/_Users_runner_work_1_s_net_newsboat/newsboat/work/newsboat-r2.20.1/target/x86_64-apple-darwin/release/build/gettext-sys-a078d908ca433957/out/lib\" -DNO_XMALLOC -Dset_relocation_prefix=libintl_set_relocation_prefix -Drelocate=libintl_relocate -DDEPENDS_ON_LIBICONV=1 -DHAVE_CONFIG_H -I. -I/opt/local/var/macports/build/_Users_runner_work_1_s_net_newsboat/newsboat/work/newsboat-r2.20.1/target/x86_64-apple-darwin/release/build/gettext-sys-a078d908ca433957/out/gettext/gettext-runtime/intl -I.. -I/opt/local/include -isysroot/Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk -O3 -ffunction-sections -fdata-sections -fPIC -Os -isysroot/Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk -arch x86_64 --target=x86_64-apple-darwin -fvisibility=hidden /opt/local/var/macports/build/_Users_runner_work_1_s_net_newsboat/newsboat/work/newsboat-r2.20.1/target/x86_64-apple-darwin/release/build/gettext-sys-a078d908ca433957/out/gettext/gettext-runtime/intl/osdep.c -o osdep.o
libtool: compile:  /usr/bin/clang -c -DLOCALEDIR=\"/opt/local/var/macports/build/_Users_runner_work_1_s_net_newsboat/newsboat/work/newsboat-r2.20.1/target/x86_64-apple-darwin/release/build/gettext-sys-a078d908ca433957/out/share/locale\" -DLOCALE_ALIAS_PATH=\"/opt/local/var/macports/build/_Users_runner_work_1_s_net_newsboat/newsboat/work/newsboat-r2.20.1/target/x86_64-apple-darwin/release/build/gettext-sys-a078d908ca433957/out/share/locale\" -DLIBDIR=\"/opt/local/var/macports/build/_Users_runner_work_1_s_net_newsboat/newsboat/work/newsboat-r2.20.1/target/x86_64-apple-darwin/release/build/gettext-sys-a078d908ca433957/out/lib\" -DBUILDING_LIBINTL -DBUILDING_DLL -DIN_LIBINTL -DENABLE_RELOCATABLE=1 -DIN_LIBRARY -DINSTALLDIR=\"/opt/local/var/macports/build/_Users_runner_work_1_s_net_newsboat/newsboat/work/newsboat-r2.20.1/target/x86_64-apple-darwin/release/build/gettext-sys-a078d908ca433957/out/lib\" -DNO_XMALLOC -Dset_relocation_prefix=libintl_set_relocation_prefix -Drelocate=libintl_relocate -DDEPENDS_ON_LIBICONV=1 -DHAVE_CONFIG_H -I. -I/opt/local/var/macports/build/_Users_runner_work_1_s_net_newsboat/newsboat/work/newsboat-r2.20.1/target/x86_64-apple-darwin/release/build/gettext-sys-a078d908ca433957/out/gettext/gettext-runtime/intl -I.. -I/opt/local/include -isysroot/Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk -O3 -ffunction-sections -fdata-sections -fPIC -Os -isysroot/Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk -arch x86_64 --target=x86_64-apple-darwin -fvisibility=hidden /opt/local/var/macports/build/_Users_runner_work_1_s_net_newsboat/newsboat/work/newsboat-r2.20.1/target/x86_64-apple-darwin/release/build/gettext-sys-a078d908ca433957/out/gettext/gettext-runtime/intl/intl-compat.c -o intl-compat.o
/bin/sh ../libtool  --tag=CC --mode=link \
	  /usr/bin/clang -I/opt/local/include -isysroot/Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk -O3 -ffunction-sections -fdata-sections -fPIC -Os -isysroot/Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk -arch x86_64 --target=x86_64-apple-darwin  -fvisibility=hidden  -L/opt/local/lib -Wl,-headerpad_max_install_names -Wl,-syslibroot,/Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk -arch x86_64 -L. -L/opt/local/lib -lsqlite3 -L/opt/local/lib -lcurl -L/opt/local/lib -lxml2 -L/opt/local/lib -lstfl -L/opt/local/lib -ljson-c -L/opt/local/lib -Wl,-search_paths_first -lncurses -framework Security -framework CoreFoundation -L/opt/local/lib -lcrypto -L/opt/local/lib -lssl -L./target/x86_64-apple-darwin/release  -o libintl.la \
	  bindtextdom.lo dcgettext.lo dgettext.lo gettext.lo finddomain.lo hash-string.lo loadmsgcat.lo localealias.lo textdomain.lo l10nflist.lo explodename.lo dcigettext.lo dcngettext.lo dngettext.lo ngettext.lo plural.lo plural-exp.lo localcharset.lo threadlib.lo lock.lo relocatable.lo langprefs.lo localename.lo log.lo printf.lo setlocale.lo version.lo xsize.lo osdep.lo intl-compat.lo -L/opt/local/lib -liconv -R/opt/local/lib -Wl,-framework -Wl,CoreFoundation   -lc \
	   \
	  -version-info 9:5:1 \
	  -rpath /opt/local/var/macports/build/_Users_runner_work_1_s_net_newsboat/newsboat/work/newsboat-r2.20.1/target/x86_64-apple-darwin/release/build/gettext-sys-a078d908ca433957/out/lib \
	  -no-undefined
make[5]: Leaving directory `/opt/local/var/macports/build/_Users_runner_work_1_s_net_newsboat/newsboat/work/newsboat-r2.20.1/target/x86_64-apple-darwin/release/build/gettext-sys-a078d908ca433957/out/build/gettext-runtime/intl'
make[4]: Leaving directory `/opt/local/var/macports/build/_Users_runner_work_1_s_net_newsboat/newsboat/work/newsboat-r2.20.1/target/x86_64-apple-darwin/release/build/gettext-sys-a078d908ca433957/out/build/gettext-runtime'
make[3]: Leaving directory `/opt/local/var/macports/build/_Users_runner_work_1_s_net_newsboat/newsboat/work/newsboat-r2.20.1/target/x86_64-apple-darwin/release/build/gettext-sys-a078d908ca433957/out/build/gettext-runtime'
make[2]: Leaving directory `/opt/local/var/macports/build/_Users_runner_work_1_s_net_newsboat/newsboat/work/newsboat-r2.20.1/target/x86_64-apple-darwin/release/build/gettext-sys-a078d908ca433957/out/build'
make[1]: Leaving directory `/opt/local/var/macports/build/_Users_runner_work_1_s_net_newsboat/newsboat/work/newsboat-r2.20.1/target/x86_64-apple-darwin/release/build/gettext-sys-a078d908ca433957/out/build'

--- stderr
configure: WARNING: unknown target-version 14, please update gt_JAVACOMP macro
make[1]: warning: -jN forced in submake: disabling jobserver mode.
../libtool: line 7472: cd: ./target/x86_64-apple-darwin/release: No such file or directory
libtool:   error: cannot determine absolute directory name of './target/x86_64-apple-darwin/release'
make[5]: *** [libintl.la] Error 1
make[4]: *** [all-recursive] Error 1
make[3]: *** [all] Error 2
make[2]: *** [all-recursive] Error 1
make[1]: *** [all] Error 2
thread 'main' panicked at '
command did not execute successfully, got: exit code: 2
```

Reply:
Not really. I'd suggest trying two things:

1. installing gettext library from the ports and supplying a path to it [like we do in Newsboat's CI pipeline](https://github.com/newsboat/newsboat/blob/e0e76c34bf3713a9e3c94d97c388120c31ad76b3/.cirrus.yml#L123-L127);
2. if macports have a port for gettext, check if it applies any patches or supplies any additional options to `configure` -- perhaps `gettext-sys` should do the same?

Reply:
> installing gettext library from the ports

The newsboat portfile already declares a dependency on the gettext port.

> and supplying a path to it [like we do in Newsboat's CI pipeline](https://github.com/newsboat/newsboat/blob/e0e76c34bf3713a9e3c94d97c388120c31ad76b3/.cirrus.yml#L123-L127);

MacPorts already sets `PATH` and adds the necessary `-I` and `-L` flags to find gettext and other libraries.

So maybe all we need to do is not list gettext-sys in cargo.crates? (Maybe the same goes for curl-sys and openssl-sys and zlib-sys, since we also have those libraries in MacPorts and would like to use them?) I'm not familiar with our rust-using ports so I don't know how that list of crates gets constructed.

Reply:
> The newsboat portfile already declares a dependency on the gettext port.

Ah, of course it does! Sorry, my bad.

> MacPorts already sets PATH and adds the necessary -I and -L flags to find gettext and other libraries.

Newsboat uses pkg-config to find its dependencies, so you might be setting yourself up for trouble if your `-I` and `-L` differ from the ones obtained via pkg-config.

Newsboat currently consists of two parts, one in C++ and the other in Rust. The compilation proceeds like this:

1. `./config.sh` finds all the libraries via pkg-config;
2. C++ part gets compiled, using `-I`/`-L` provided by pkg-config. These can be overriden by you via `CXXFLAGS`;
3. Rust part gets compiled. Each crate runs pkg-config by itself. You can only override this via environment variables related to pkg-config, or via crate-specific variables;
4. everything gets linked using flags obtained from pkg-config, which you can override via `LDFLAGS`.

Steps 2 and 3 run in parallel, of course, but that's beside the point.

I hope this explains why you need to set additional environment variables to affect `gettext-sys`.

> So maybe all we need to do is not list gettext-sys in cargo.crates? (Maybe the same goes for curl-sys and openssl-sys and zlib-sys, since we also have those libraries in MacPorts and would like to use them?)

I'm not sure, but my understanding is that by removing those crates from the port, you'll force Cargo to download the crates from the Internet, which would probably fail because you probably disallow Internet access during builds.

As far as I understand, this build failure only occurs with `gettext-sys` because:

1. macOS doesn't ship gettext as part of libc;
2. gettext library doesn't provide a pkg-config file, so `gettext-sys` can't find the library automatically (like `curl-sys` and `openssl-sys` do with their respective libraries).

This forces `gettext-sys` to try and build its own gettext library on macOS. All the other `-sys` crates find their corresponding libraries via pkg-config and thus don't cause any problems.